packages/firestore/src/util/sorted_map.ts

```diff
@@ -298,14 +301,14 @@ export class SortedMapIterator<K, V, T> {
     }
   }
 
-  getNext(): T {
+  getNext(): T | Entry<K, V> {
```

Comment:
This is a pretty weird return type and I was surprised it didn't cause you to have to cast in all of the usages of it, but I _think_ T always ends up being Entry<K, V> (because we don't provide a resultGenerator) and so the type simplifies to just Entry<K, V> and no cast is necessary.

But I think I would just remove resultGenerator entirely and then the return type here is just Entry<K, V>. Android and iOS don't seem to have a resultGenerator anyway...

Reply:
I was debating getting rid of resultGenerator, but wasn't certain where this code was coming from and whether we wanted to keep it closer to the original implementation. But I am going to remove it since it will make the types much cleaner.